google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherSnippets.java

```diff
@@ -54,24 +67,112 @@ public void onFailure(Throwable t) {
       }
     });
-    // [END publish]
     return messageIdFuture;
   }
 
   /** Example of creating a {@code Publisher}. */
   // [TARGET newBuilder(TopicName)]
   // [VARIABLE "my_project"]
   // [VARIABLE "my_topic"]
-  public static void newBuilder(String projectName, String topicName) throws Exception {
-    // [START newBuilder]
-    TopicName topic = TopicName.create(projectName, topicName);
+  public static void newBuilder(String projectId, String topicId) throws Exception {
+    TopicName topic = TopicName.create(projectId, topicId);
     Publisher publisher = Publisher.defaultBuilder(topic).build();
     try {
       // ...
     } finally {
       // When finished with the publisher, make sure to shutdown to free up resources.
       publisher.shutdown();
     }
-    // [END newBuilder]
+  }
+
+  public Publisher getPublisherWithCustomBatchSettings(TopicName topicName) throws Exception {
+    // [START publisherBatchSettings]
+    // Batch settings control how the publisher batches messages
+    long requestBytesThreshold = 5000L; // default : 1kb
+    long messageCountBatchSize = 10L; // default : 100
+
+    Duration publishDelayThreshold = Duration.ofMillis(100); // default : 1 ms
+
+    // Batching thresholds have order precedence : request size  > message count > publish delay
+    BatchingSettings batchingSettings = BatchingSettings.newBuilder()
+        .setElementCountThreshold(messageCountBatchSize)
+        .setRequestByteThreshold(requestBytesThreshold)
+        .setDelayThreshold(publishDelayThreshold)
+        .build();
+
+    Publisher publisher = Publisher.defaultBuilder(topicName)
+        .setBatchingSettings(batchingSettings).build();
+    // [END publisherBatchSettings]
+    return publisher;
+  }
+
+  public Publisher getPublisherWithCustomRetrySettings(TopicName topicName) throws Exception {
+    // [START publisherRetrySettings]
+    // Retry settings control how the publisher handles retryable failures
+    Duration retryDelay = Duration.ofMillis(100); // default : 1 ms
+    double retryDelayMultiplier = 2.0; // back off for repeated failures
+    Duration maxRetryDelay = Duration.ofSeconds(5); // default : 10 seconds
+
+    RetrySettings retrySettings = RetrySettings.newBuilder()
+        .setInitialRetryDelay(retryDelay)
+        .setRetryDelayMultiplier(retryDelayMultiplier)
+        .setMaxRetryDelay(maxRetryDelay)
+        .build();
+
+    Publisher publisher = Publisher.defaultBuilder(topicName)
+        .setRetrySettings(retrySettings).build();
+    // [END publisherRetrySettings]
+    return publisher;
+  }
+
+  public Publisher getPublisherWithCustomFlowControlSettings(TopicName topicName) throws Exception {
+    // [START publisherFlowControlSettings]
+
+    // Flow control settings restrict the number of outstanding publish requests
+    int maxOutstandingBatches = 20;
+    int maxOutstandingRequestBytes = 500000;
+
+    // override behavior on limits exceeded, default behavior is to throw an exception
```

Comment:
I could have sworn I saw default as throw exception :) corrected. thanks for catching.